hbase-common/src/main/java/org/apache/hadoop/hbase/util/DNS.java

```diff
@@ -35,13 +35,21 @@
   // the specification of server hostname is optional. The hostname should be resolvable from
   // both master and region server
   @InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.CONFIG)
-  public static final String RS_HOSTNAME_KEY = "hbase.regionserver.hostname";
+  public static final String UNSAFE_RS_HOSTNAME_KEY = "hbase.unsafe.regionserver.hostname";
   @InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.CONFIG)
   public static final String MASTER_HOSTNAME_KEY = "hbase.master.hostname";
 
   private static boolean HAS_NEW_DNS_GET_DEFAULT_HOST_API;
   private static Method GET_DEFAULT_HOST_METHOD;
 
+  /**
+   * @deprecated since 2.4.0 and in 3.0.0, to be removed in 4.0.0.
```

Comment:
It's fine to only mention the first release it gets deprecated. It would also be great if you could add the link to the deprecating ticket via a `@see` annotation (you'll find an example [here](https://github.com/ultratendency/hbase/blob/74d37eace9218d626586e8afee751eb52bbe5311/hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java#L1363)).

Reply:
Not sure if I followed all discussions. This one will be backported to branch-2, right? If so, rephrase it to "since 2.4.0. Will be removed in 4.0.0." as 2.4.0 is the first release of deprecation. Same applies to my other comment.